### PR TITLE
Add deviceid example

### PR DESCRIPTION
**Pull request by dbrgn**
First attempt, as requested here: https://github.com/japaric/particle-hal/pull/6

### CODE REVIEW

File: examples/deviceid.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+use hal::delay;
+use hal::usb;
+use hal::deviceid;
+use numtoa::NumToA;
+
+const PERIOD: u32 = 1000; // ms
+const BAUD_RATE: u32 = 115_200;
+
+static mut DEVICE_ID: [u8; 12] = [0; 12];
```
Comment (japaric) [inline examples/deviceid.rs]: These variables could be allocated on the `loop_`'s stack. `let mut device_id: [0; 12]`.

**Comment by dbrgn** [comment]
Amended.

**Comment by japaric** [comment]
Thanks!

@homunkulus r+

**Comment by homunkulus** [comment]
:pushpin: Commit 74d24bb has been approved by `japaric`

**Comment by homunkulus** [comment]
:hourglass: Testing commit 74d24bb with merge 74d24bb...

**Comment by homunkulus** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/japaric/photon-quickstart/builds/231905638?utm_source=github_status&utm_medium=notification)
Approved by: japaric
Pushing 74d24bbb921074437e16ccc49872b64f0ecdec01 to master...

**Pull request merged**